lab-christian/lib/basic-auth-middleware.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const createError = require('http-errors');
+const debug = require('debug')('instaclone:basic-auth');
+
+module.exports = function(req, res, next) {
+  debug('basic auth');
+  
+  var authHeader = req.headers.authorization;
+  if (!authHeader) return next(createError(401, 'authorization header required'));
+
+
+  var base64str = authHeader.split('Basic ')[1];
+  if (!base64str) return next(createError(401, 'username and password required'));
+
+  var utf8str = Buffer.from(base64str, 'base64').toString();
+  var authArray = utf8str.split(':');
```

Comment:
check it out:

`const [username, password] = utf8str.split(':');`

And now ya got em and can use em.